experiment/simulation/js/main.js

```diff
@@ -1 +1,778 @@
+	let step, translate, rotation, lim, rotLim, objs, keys, enabled, small, cutStep, hitCtr;
+	init();
+
+	const tableData = [
+		{ "Soil Sample No.": "", "Water Content(%)": "", "Wet Compacted Soil Mass(g)": "", "Wet Density(g/cc)": "", "Dry Density(g/cc)": "" }, 
+		{ "Soil Sample No.": "", "Water Content(%)": "", "Wet Compacted Soil Mass(g)": "", "Wet Density(g/cc)": "", "Dry Density(g/cc)": "" }, 
+		{ "Soil Sample No.": "", "Water Content(%)": "", "Wet Compacted Soil Mass(g)": "", "Wet Density(g/cc)": "", "Dry Density(g/cc)": "" }, 
+		{ "Soil Sample No.": "", "Water Content(%)": "", "Wet Compacted Soil Mass(g)": "", "Wet Density(g/cc)": "", "Dry Density(g/cc)": "" }, 
+		{ "Soil Sample No.": "", "Water Content(%)": "", "Wet Compacted Soil Mass(g)": "", "Wet Density(g/cc)": "", "Dry Density(g/cc)": "" }, 
+	];
+
+	const objNames = Object.keys(objs);
+	objNames.forEach(function(elem, ind) {
+		const obj = document.getElementById(elem);
+		obj.addEventListener('click', function(event) {
+			if(elem === "soil")
+			{
+				soilVol = randomNumber(940, 950);
+				document.getElementById("output2").innerHTML = "Volume of soil = " + String(soilVol) + "cm" + "3".sup();
+			}
+
+			keys.push(elem);
+			step += 1;
+		});
+	});
+
+	canvas.addEventListener('mousemove', function(event) {check(event, translate, step, false);});
+	canvas.addEventListener('click', function(event) {
+		step = check(event, translate, step);
+	});
+
+	const table = document.getElementsByClassName("table")[0];
+
+	function responsiveTable(x) {
+		if(x.matches)	// If media query matches
+		{ 
+			small = true;
+			if(step === enabled.length - 1)
+			{
+				document.getElementById("observations").style.width = '85%';
+			}
+		} 
+
+		else
+		{
+			small = false;
+			if(step === enabled.length - 1)
+			{
+				document.getElementById("observations").style.width = '40%';
+			}
+		}
+	};
+
+	let x = window.matchMedia("(max-width: 1023px)");
+	responsiveTable(x); // Call listener function at run time
+	x.addListener(responsiveTable); // Attach listener function on state changes
+
+	function draw()
+	{
+		ctx.clearRect(0, 0, canvas.width, canvas.height); 
+		ctx.lineCap = "round";
+		ctx.lineJoin = "round";
+
+		let ctr = 0;
+		document.getElementById("main").style.pointerEvents = 'none';
+
+		objNames.forEach(function(name, ind) {
+			document.getElementById(name).style.pointerEvents = 'auto';
+			if(keys.includes(name) || !(enabled[step].includes(name)))
+			{
+				document.getElementById(name).style.pointerEvents = 'none';
+			}
+
+			if(keys.includes(name)) 
+			{
+				if(enabled[step].includes(name))
+				{
+					ctr += 1;
+				}
+				objs[name].draw(ctx);
+			}
+		});
+
+		if(ctr === enabled[step].length && !rotation &&!translate[0] && !translate[1])
+		{
+			document.getElementById("main").style.pointerEvents = 'auto';
+		}
+
+		if(step === 6 && rotation && rotate([objs['water']], rotation, rotLim))
+		{
+			if(objs['water'].flow(0.01))
+			{
+				objs['soil'].color = "#b86d29";
```

Comment:
Consider moving all color codes into an enum in a separate js file with mnemonic keys and hex code values